[Book][Forms] describe the allow_extra_fields form option

xabbuh opened the pull request:
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes (symfony/symfony#8133) |
| Applies to | 2.6+ |
| Fixed tickets | #4082 |

xabbuh committed:
describe the allow_extra_fields form option

WouterJ commented:
@xabbuh, as you know there will be a sprint day soon :wink:. Do you have time to update this PR before/during that day or should I put it on the list?

xabbuh commented:
Sorry for the long delay. I updated the section and also moved it to the "Validation" part of the Form chapter (I think it better fits in that context).

weaverryan commented on book/forms.rst:
```diff
@@ -625,6 +625,21 @@ Now the form will skip your validation constraints. It will still validate
 large or whether you tried to submit text in a number field.
 
+Handling Additional Fields
+~~~~~~~~~~~~~~~~~~~~~~~~~~
```
How about: Handle Extra Fields

weaverryan commented on book/forms.rst:
```diff
@@ -625,6 +625,21 @@ Now the form will skip your validation constraints. It will still validate
+.. versionadded:: 2.6
+    The ``allow_extra_fields`` option was introduced in Symfony 2.6.
+
+Usually, the Form component will complain when your submitted form contains
```
How about this (I wanted to change from "your submitted form" to making "submit" sound more like a verb - since I understand that I "submit" data, but the definition of what a "submitted form" is, is a little less clear I think):

>
Usually, if you submit extra fields that aren't configured in your form, you'll get a
"This form should not contain extra fields." validation error (for example ...)

weaverryan commented on book/forms.rst:
```diff
@@ -625,6 +625,21 @@ Now the form will skip your validation constraints. It will still validate
+
+Usually, the Form component will complain when your submitted form contains
+fields that were not configured before (for example, when fields are added
+using JavaScript or when a page contains several forms without a name).
```
Multiple forms without a name? Would that cause an issue? You can only submit *one* of the forms - it would only be an issue if you made them submit to the same endpoint. That's possible - is it something that really happens?

xabbuh replied:
Not sure how often you really want to submit multiple nameless forms to the same endpoint (this was a reason given in symfony/symfony#8133). However, you will probably notice when you need this option. So I decided to remove this part.

weaverryan commented on book/forms.rst:
```diff
@@ -625,6 +625,21 @@ Now the form will skip your validation constraints. It will still validate
+fields that were not configured before (for example, when fields are added
+using JavaScript or when a page contains several forms without a name).
+
+You can silence this error message by enabling the ``allow_extra_fields``
```
... silence this validation error by ...

That way it's clear we're fixing the validation problem, not just silencing the error

weaverryan commented on book/forms.rst:
```diff
@@ -625,6 +625,21 @@ Now the form will skip your validation constraints. It will still validate
+option on the form::
+
+    $form = $this->createForm(new TaskType(), $task, array('allow_extra_fields' => true));
+
```
Should we add this to the `form.rst` reference file? In fact, I think we should perhaps move this whole thing there - keep the form chapter shorter?

xabbuh replied:
I think you are right. I moved it to the reference of the ``form`` type.

weaverryan commented:
Yes! Thanks @xabbuh!
